### PR TITLE
[CAPI/acr] Update tizen capi header

### PR DESCRIPTION
**Pull request by zhoonit**
**Changes proposed in this PR:**
- Setting void *user_data
- Update layer type enum

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

### CODE REVIEW

File: api/nntrainer-api-common.h [hunk 1]
```diff
@@ -27,8 +27,15 @@ extern "C" {
...
-  ML_TRAIN_LAYER_TYPE_FC,           /**< Fully Connected Layer */
+  ML_TRAIN_LAYER_TYPE_INPUT = 0,  /**< Input Layer */
+  ML_TRAIN_LAYER_TYPE_FC,         /**< Fully Connected Layer */
+  ML_TRAIN_LAYER_TYPE_BN,         /**< Batch Normalization Layer (since 6.5) */
+  ML_TRAIN_LAYER_TYPE_CONV2D,     /**< Convolution 2D Layer (since 6.5) */
+  ML_TRAIN_LAYER_TYPE_POOLING2D,  /**< Pooling 2D Layer (since 6.5) */
+  ML_TRAIN_LAYER_TYPE_FLATTEN,    /**< Flatten Layer (since 6.5) */
+  ML_TRAIN_LAYER_TYPE_ACTIVATION, /**< Activation Layer (since 6.5) */
+  ML_TRAIN_LAYER_TYPE_ADDITION,   /**< Addition Layer (since 6.5) */
+  ML_TRAIN_LAYER_TYPE_CONCAT,     /**< Concat Layer (since 6.5) */
```
Comment (kparichay) [inline api/nntrainer-api-common.h:38]: Please add nnstreamer and tflite backbone layer as well.

File: api/capi/include/nntrainer.h [hunk 2]
```diff
@@ -357,6 +357,34 @@ int ml_train_dataset_create_with_generator(ml_train_dataset_h *dataset,
...
+ * @param[in] valid_user_data user data to be passed when @a valid_cb is being
+ * called. Can be null.
+ * @param[in] test_user_data user data to be passed when @a test_cb is being
+ * called. Can be null.
+ * @return @c 0 on successs. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successsful
+ * @retval #ML_ERROR_INVALID_PARAMETER if @a dataset is not created from @a
+ * ml_train_dataset_create_with_generator
+ */
+int ml_train_dataset_generator_set_user_data(ml_train_dataset_h *dataset,
```
Comment (kparichay) [inline api/capi/include/nntrainer.h:384]: If you are adding this, please revert the changes from https://github.com/nnstreamer/nntrainer/pull/533

Comment (zhoonit) [reply api/capi/include/nntrainer.h:384]: Agh.. I forgot we had that functionality already.

I think #533 style is better, in this case, we just have to change the document.

I wonder if we can split those into train_user_data, valid_user_data, ... as far as I remember there was a problem with this.
"ml_train_dataset_set_property(dataset, "train_user_data", (void *) data, " NULL)"

Comment (kparichay) [reply api/capi/include/nntrainer.h:384]: Yes, we have to support 3 user data corresponding to each train, valid and test.
That will have to added w.r.t. #533.

File: api/nntrainer-api-common.h (hunk 1 above)
Comment (zhoonit) [reply api/nntrainer-api-common.h]: We discussed offline to skip this change

File: api/capi/include/nntrainer.h (hunk 2 above)
Comment (zhoonit) [reply api/capi/include/nntrainer.h]: We discussed offline to add something like "train_user_data", "validation_user_data", "test_user_data" w.r.t #533

File: api/nntrainer-api-common.h [hunk 3]
```diff
@@ -27,8 +27,15 @@ extern "C" {
  * @since_tizen 6.0
  */
 typedef enum {
-  ML_TRAIN_LAYER_TYPE_INPUT = 0,    /**< Input Layer */
-  ML_TRAIN_LAYER_TYPE_FC,           /**< Fully Connected Layer */
+  ML_TRAIN_LAYER_TYPE_INPUT = 0,  /**< Input Layer */
+  ML_TRAIN_LAYER_TYPE_FC,         /**< Fully Connected Layer */
```
Comment (kparichay) [inline api/nntrainer-api-common.h:31]: you can consider making it alphabetical.

**Review by kparichay — APPROVED** [review]
LGTM

File: api/nntrainer-api-common.h (hunk 3 above)
Comment (zhoonit) [reply api/nntrainer-api-common.h:31]: Input and FC is released so I guess it is not possible?

Comment (kparichay) [reply api/nntrainer-api-common.h:31]: We do not any applications with nntrainer, so IMO it can be done.
Please take advice from @again4you on this.

Comment (zhoonit) [reply api/nntrainer-api-common.h:31]: If we keep making it sorted, this will break someday let alone additional ACR we have to go thorugh.
As we are going to keep adding layers. Let's keep it as it is if there's no good reason.

Comment (kparichay) [reply api/nntrainer-api-common.h:31]: works.

File: api/ccapi/include/layer.h [hunk 4]
```diff
@@ -33,17 +33,18 @@ namespace train {
...
-  LAYER_BACKBONE_TFLITE,                /** Backbone using TFLite */
+  LAYER_BN = ML_TRAIN_LAYER_TYPE_BN,    /** Batch Normalization Layer type */
+  LAYER_CONV2D = ML_TRAIN_LAYER_TYPE_CONV2D, /** Convolution 2D Layer type */
+  LAYER_POOLING2D = ML_TRAIN_LAYER_TYPE_POOLING2D, /** Pooling 2D Layer type */
+  LAYER_FLATTEN = ML_TRAIN_LAYER_TYPE_FLATTEN,     /** Flatten Layer type */
+  LAYER_ACTIVATION =
+    ML_TRAIN_LAYER_TYPE_ACTIVATION,              /** Activation Layer type */
+  LAYER_ADDITION = ML_TRAIN_LAYER_TYPE_ADDITION, /** Addition Layer type */
+  LAYER_MULTIOUT = ML_TRAIN_LAYER_TYPE_MULTIOUT, /** Multi Output Layer type */
+  LAYER_CONCAT,                                  /** Concat Layer type */
```
Comment (jijoongmoon) [inline api/ccapi/include/layer.h:44]: We need this concat. resnet needs it.

Comment (zhoonit) [reply api/ccapi/include/layer.h:44]: Updated

**Review by jijoongmoon — APPROVED** [review]
LGTM.
